--- cvsnt-2.5.04.2980/configure.in~	2008-02-14 06:26:00.000000000 +0200
+++ cvsnt-2.5.04.2980/configure.in	2008-03-27 00:06:32.828790347 +0200
@@ -24,8 +24,7 @@
 CXXFLAGS="$CXXFLAGS $OPTFLAGS"
 
 AC_LIBTOOL_DLOPEN
-AC_LIBLTDL_CONVENIENCE
-AC_CONFIG_SUBDIRS(libltdl)
+LIBLTDL="-lltdl"
 AC_SUBST(INCLTDL)
 AC_SUBST(LIBLTDL)
 
--- cvsnt-2.5.04.2980/Makefile.am	2008-03-27 00:06:10.903530846 +0200
+++ cvsnt-2.5.04.2980/Makefile.am	2008-03-27 00:14:19.591605118 +0200
@@ -38,8 +38,6 @@
   RCSTOOLS_SUBDIR =
 endif
 
-ltdl_subdir = libltdl
-
 if WITH_MAC_HFS_SUPPORT
   MACOSX_SUBDIR = osx
 else
@@ -55,7 +53,7 @@
 cvs_config_dir=@cvs_config_dir@
 
 # All other subdirs:
-SUBDIRS = $(ZLIB_SUBDIR) lib cvsapi/ufc-crypt $(ltdl_subdir) $(EXPAT_SUBDIR) $(PCRE_SUBDIR) $(MDNSCLIENT_SUBDIR) cvsapi diff cvsgui cvstools xdiff \
+SUBDIRS = $(ZLIB_SUBDIR) lib cvsapi/ufc-crypt $(EXPAT_SUBDIR) $(PCRE_SUBDIR) $(MDNSCLIENT_SUBDIR) cvsapi diff cvsgui cvstools xdiff \
 	  protocols triggers cvsdelta $(MACOSX_SUBDIR) src $(LOCKSERVER_SUBDIR) $(RCSTOOLS_SUBDIR) man run_script
 
 EXTRA_DIST = \
