--- cvsnt-2.5.04.2980/configure.in~	2008-03-27 00:22:45.000000000 +0200
+++ cvsnt-2.5.04.2980/configure.in	2008-03-27 00:29:40.212501593 +0200
@@ -1004,7 +1004,6 @@
 AM_CONDITIONAL(SSPI_PROTOCOL_SUPPORT, test $acx_sspi = "yes")
 if test x$acx_sspi = xyes; then
   AC_DEFINE(SSPI_PROTOCOL_SUPPORT, [], [Support for :sspi:])
-  AC_CONFIG_SUBDIRS(protocols/ntlm)
 fi
 AC_MSG_RESULT($acx_sspi)
 
--- cvsnt-2.5.04.2980/protocols/Makefile.am~	2006-08-18 00:10:05.000000000 +0300
+++ cvsnt-2.5.04.2980/protocols/Makefile.am	2008-03-27 00:35:53.073927679 +0200
@@ -79,9 +79,10 @@
 	sspi_unix.cpp \
 	common.cpp 
 
-sspi_la_LDFLAGS = -module -avoid-version  -L$(DESTDIR)/$(libdir) ntlm/libntlm.la
+sspi_la_LDFLAGS = -module -avoid-version  -L$(DESTDIR)/$(libdir)
+sspi_la_LIBADD = -lntlm
 
-sspi_ntlm = ntlm
+sspi_ntlm = 
 
 else
 
