--- cvsnt-2.5.05.3744/configure.in~	2009-11-24 15:52:22.000000000 +0100
+++ cvsnt-2.5.05.3744/configure.in	2010-04-10 12:34:00.151348422 +0200
@@ -370,10 +370,11 @@
 
 # Check for pcre posix library
 saved_libs="$LIBS"
-LIBS=""
-AC_SEARCH_LIBS(pcre_compile, pcreposix, [AC_MSG_CHECKING([for PCRE posix library])
-     AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <pcreposix.h>]], [[int i;]])],[ac_cv_have_pcre="yes"],[])
-     AC_MSG_RESULT($ac_cv_have_pcre)], ac_cv_have_pcre="no")
+LIBS="-lpcreposix"
+AC_MSG_CHECKING([for PCRE posix library])
+AC_TRY_LINK([#include <pcreposix.h>], [regcomp(0, 0, 0);],
+	[ac_cv_have_pcre="yes"],[ac_cv_have_pcre="no"])
+AC_MSG_RESULT([$ac_cv_have_pcre])
 PCRE_LIBS="$LIBS"
 LIBS="$saved_libs"
 AC_SUBST(PCRE_LIBS)
